stacker/blueprints/base.py

```diff
@@ -239,63 +187,63 @@ def __init__(self, name, context, mappings=None):
         self.context = context
         self.mappings = mappings
         self.outputs = {}
-        self.local_parameters = self.get_local_parameters()
         self.reset_template()
         self.resolved_variables = None
 
-    @property
-    def parameters(self):
-        return self.template.parameters
-
-    @property
-    def required_parameters(self):
+    def get_required_parameters(self):
         """Returns all template parameters that do not have a default value."""
         required = []
-        for k, v in self.parameters.items():
-            if not hasattr(v, "Default"):
-                required.append((k, v))
+        for name, attrs in self.template.parameters.iteritems():
+            if not hasattr(attrs, "Default"):
+                required.append((name, attrs))
         return required
 
-    def get_local_parameters(self):
-        local_parameters = getattr(self, "LOCAL_PARAMETERS", {})
-        return get_local_parameters(local_parameters, self.context.parameters)
+    def get_parameter_definitions(self):
+        """Get the parameter definitions to submit to CloudFormation.
 
-    def _get_parameters(self):
-        """Get the parameter definitions.
-
-        First looks at CF_PARAMETERS, then falls back to PARAMETERS for
-        backwards compatibility. This will also return any variables whose
-        `type` is an instance of `CFNType`.
-
-        Makes this easy to override going forward for more backwards
-        compatibility.
+        Any variable definition whose `type` is an instance of `CFNType` will
+        be returned as a CloudFormation Parameter.
 
         Returns:
             dict: parameter definitions. Keys are parameter names, the values
-                are dicts containing key/values for various parameter
-                properties.
-        """
-        parameters = getattr(self, "CF_PARAMETERS",
-                             getattr(self, "PARAMETERS", {}))
+            are dicts containing key/values for various parameter properties.
 
+        """
+        output = {}
         for var_name, attrs in self.defined_variables().iteritems():
             var_type = attrs.get("type")
             if isinstance(var_type, CFNType):
                 cfn_attrs = copy.deepcopy(attrs)
                 cfn_attrs["type"] = var_type.parameter_type
-                parameters[var_name] = cfn_attrs
-        return parameters
+                output[var_name] = cfn_attrs
+        return output
+
+    def get_parameters(self):
+        """Return a dictionary of variables with `type` :class:`CFNType`.
+
+        Returns:
+            dict: variables that need to be submitted as CloudFormation
+                Parameters.
+
+        """
+        variables = self.get_variables()
+        output = {}
+        for key, value in variables.iteritems():
+            if hasattr(value, "to_parameter_value"):
```

Comment:
Should probably replace this with:

```
try:
    output[key] = value.to_parameter_value()
except AttributeError:
    continue
```

It's a nit, but doing hasattr doesn't check that `to_parameter_value()` is actually callable.